imap/jmap_sieve.c

```diff
@@ -456,8 +456,37 @@ static int jmap_sieve_get(jmap_req_t *req)
     return 0;
 }
 
-static sieve_script_t *parsescript(FILE *f, const char *content, char **errors)
+static int putscript(const char *name, const char *content,
+                     const char *sievedir, json_t **err)
 {
+    char new_path[PATH_MAX];
+    FILE *f;
+
+    /* parse the script */
+    char *errors = NULL;
+    sieve_script_t *s = NULL;
+    (void) sieve_script_parse_string(NULL, content, &errors, &s);
+    if (!s) {
+        *err = json_pack("{s:s, s:s}", "type", "invalidScript",
+                         "description", errors);
+        free(errors);
+        return 0;
+    }
+
+    /* open a new file for the script */
+    snprintf(new_path, sizeof(new_path),
+             "%s/%s%s.NEW", sievedir, name, SCRIPT_SUFFIX);
+
+    f = fopen(new_path, "w+");
+
+    if (f == NULL) {
+        syslog(LOG_ERR, "fopen(%s): %m", new_path);
```

Comment:
Does this log line want an IOERROR label on it?  I notice that we didn't previously have one in the old location, but I'm not sure if that was deliberate or oversight.